doc/fault_management/fault_mgmt_infra_HLD.md

```diff
@@ -0,0 +1,173 @@
+   2) Fetch these events (alarms/faults) from the eventD (based on published YANG/schema)
+   3) Analyze them (in a generic way) against the above-mentioned Policy Table
+   4) Take action based on the lookup/match in Policy Table
+   5) Action could either be generic or platform specific
+
+# Objective
+Objective of producing this document is two-fold:
+
+   a) Every SONiC NOS deployment may not have External Controller to take the action upon fault occurence. 
+      In that case, SONiC (with its underlying platform) is expected to take the required action to recover the system/chassis from the fault.
+      
+   b) Platform supplied 'Fault-Action Policy table' has a holistic/system-level view of the platform (chassis/board/HWSKU) 
+      and can gauge the right action required to recover the system from the fault.
+      It can either go with the recommended action (provided by the FDR - fault source/detector) or override it with the system-level one.
+
+**Fault Manager** module (as described in below block diagram) would serve the purpose of taking necessary action(s) to log and handle the faults.
+
+# High Level Block diagram and System Flow
+
+High-Level design (end-to-end workflow) to inject, detect, report, analyze and handle Faults in a SONiC based chassis.
+
+Note: Errors/Failures/Faults are regarded as Faults here.
+
+<img width="892" alt="Screenshot 2023-11-30 at 5 36 53 PM" src="https://github.com/shyam77git/SONiC/assets/69485234/47a5fa4a-7f3d-4d45-86a4-81cb3fb893f1">
+
+**Block diagram's markers/steps (#1 through #9) explanation:**
+|      Markers        |                   Brief Description                                    | State                                                  |
+| --------------------|------------------------------------------------------------------------|--------------------------------------------------------|
+| Markers 1 through 3 | Fault detector (source) publishes its events(faults, alarms) to eventD | Available in SONiC 202305
+| Marker 4            | Storing the events to eventD's redisDB instance                        | codeflow to be committed as part of revised HLD: https://github.com/sonic-net/SONiC/pull/1409 |                                                                            
+| Marker 5 through 9  | This is the beige oval-shaped box (on the right) representing the FM (Fault Managment) infrastructure module being added and developed as part of this HLD/Feature| This HLD|
+                                                                    
+
+# High Level Work Flow
+Following are the main functionalities/tasks of the **FM infrastructure module** being introduced as part of this document:
+1. Formulate platform/HWSKU specific Fault-Action Policy Table (json or yaml file)
+   - There would be generic (default) table if none provided by platform
+   - A platform supplied file would override the default one
+2. Introduce a new micro-service (fault_manager) at host (Linux Kernel)
```

Comment:
What is the plan? Is the fault manager a dedicated docker container or service? or it it colocated with EventD docker container?